### PR TITLE
Bops debug

### PR DESCRIPTION
**Pull request by homemmakako**
Adds a custom logger to meta-system and fixes a few bugs

### CODE REVIEW

File: src/common/logger/get-system-info.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import OS from "os";
```
Comment (zelcion) [inline src/common/logger/get-system-info.ts:1]: We might move this into another folder, since this info may not be used only in the logger

**Review by zelcion — APPROVED** [review]
Awesome overall. Great Work

File: src/bin/index.ts [hunk 2]
```diff
@@ -1,39 +1,67 @@
...
-    console.log(`Running meta-system version ${packageData.version}`);
-    return;
-  }
+const program = new Command("meta-system");
+program
+  .showSuggestionAfterError(true)
+  .helpOption("-h, --help", "Displays this help panel")
+  .version("Currently on version " + packageData.version, "-v, --version", "Displays the current meta-system version")
+  .option("-d, --debug", "Logs additional info on the execution of BOps", () => {
+    environment.silent.constants.logLevel = "debug";
```
[the remark this answers is not in the document]
Comment (homemmakako) [reply src/bin/index.ts:20]: Silent simply makes it so nothing will be logged. By default, trying to reset/delete a constant will result in an error, informed through stdout. If you use silent constants however no error message will be shown. 
It should be noted that both map to the same object, so `environment.constants` and `environment.silent.constants` are the same, only the attribuition functions are diferent. You can think of this much like the `expect` sintax where you can concatenate words to chage the behavior slightly (`expect().to.be.undefined` vs `expect().not.to.be.undefined` for instance)

File: src/bops-functions/function-managers/external-function-manager.ts [hunk 3]
```diff
@@ -5,16 +5,17 @@ import { runtimeDefaults } from "../../configuration/runtime-config/defaults";
...
   private functionMap : Map<string, Function> = new Map();
   private infoMap : Map<string, MetaFunction> = new Map();
 
   public constructor (
-    private functionsInstaller = new FunctionsInstaller(runtimeDefaults.externalFunctionInstallFolder),
+    private functionsInstaller = new FunctionsInstaller(environment.constants.installDir),
     private functionFileSystem = new FunctionFileSystem(
-      runtimeDefaults.externalFunctionInstallFolder,
+      environment.constants.installDir,
       runtimeDefaults.externalFunctionConfigFileName,
```
[the remark this answers is not in the document]
Comment (homemmakako) [reply src/bops-functions/function-managers/external-function-manager.ts:19]: At first I'd like to keep them separated as there is a fundamental difference between them:
- environment constants do not change during runtime, but they do change depending on the context (system,machine,OS,etc)
- runtimeDefaults are always constant, no matter the context

That said, I see no problem with having them as a immutable part of environment, maybe we could save them as object `environment.defaults` (e.g.: `environment.defaults.logLevel` for the default log level)? What do you think ?

**Pull request merged**